### ISSUE TITLE
Term1 - Lesson 'Vehicle Detection and Tracking' - 28. Color Classify - Data snooping

### ISSUE THREAD

**Issue by bdelhaisse** [udacity/sdc-issue-reports#814]
In the lesson 28. 'Color Classify', in 'Vehicle Detection and Tracking' (Term1), in the explanations and quiz, you are doing data snooping.
Basically, you are splitting the data into training and test set after having rescaled the data using StandardScaler. This is an error as you are using some data points from the test set to rescale your data, and thus the test set is not really a test set anymore. You should first split the data, and then rescale the training and test set but using only the training set to scale.
In code:
```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=rand_state)
X_train_scaler = StandardScaler().fit(X_train)
scaled_X_train = X_train_scaler.transform(X_train)
scaled_X_test = X_train_scaler.transform(X_test)
```

**Comment by karlobermeyer** [comment]
Nice catch. I also noticed this problem  in "Vehicle Detection and Tracking" part 34 "Search and Classify", the file `search_classify.py` line 174, as provided to students for the quiz.

**Comment by mvirgo** [comment]
Great catch by both of you! I have updated the classroom to change the scaler to come after `train_test_split` in both of these lessons now. Looks like "Hog Classify" did this as well.

I also added a note on this to the "Combine and Normalize Features" lesson, so that students know this is important when it first comes up.
